django/db/models/fields/related_lookups.py

```diff
@@ -9,22 +10,37 @@
 )
 
 
-class MultiColSource:
+class MultiColSource(Expression):
     contains_aggregate = False
     contains_over_clause = False
 
     def __init__(self, alias, targets, sources, field):
-        self.targets, self.sources, self.field, self.alias = (
-            targets,
-            sources,
-            field,
-            alias,
-        )
-        self.output_field = self.field
+        super().__init__(output_field=field)
+        self.targets, self.sources, self.alias = targets, sources, alias
 
     def __repr__(self):
         return "{}({}, {})".format(self.__class__.__name__, self.alias, self.field)
 
+    def __len__(self):
+        return len(self.targets)
+
+    def get_source_expressions(self):
+        return [Col(self.alias, target) for target in self.targets]
+
+    def set_source_expressions(self, exprs):
+        assert all(isinstance(expr, Col) for expr in exprs)
+        assert len(exprs) == len(self.targets)
+
+    def as_sql(self, compiler, connection):
+        sqls = []
+        cols = self.get_source_expressions()
+
+        for col in cols:
+            sql, _ = col.as_sql(compiler, connection)
```

Comment:
What is `_` here? Should we be doing something with it?

Reply:
It's the `params`, it's safe to ignore because it's always a `[]`.

Reply:
Okay, sure 👍